packages/gatsby-cli/src/plugin-add.ts

```diff
@@ -0,0 +1,83 @@
+import { NPMPackage, GatsbyPlugin } from "gatsby-recipes"
+import reporter from "./reporter"
+const normalizePluginName = (plugin: string): string => {
+  if (plugin.startsWith(`gatsby-`)) {
+    return plugin
+  }
+  if (
+    plugin.startsWith(`source-`) ||
+    plugin.startsWith(`transformer-`) ||
+    plugin.startsWith(`plugin-`)
+  ) {
+    return `gatsby-${plugin}`
+  }
+  return `gatsby-plugin-${plugin}`
+}
+
+async function installPluginPackage(
+  plugin: string,
+  root: string
+): Promise<void> {
+  const installTimer = reporter.activityTimer(`Installing ${plugin}`)
+
+  installTimer.start()
+  reporter.info(`Installing ${plugin}`)
+  try {
+    const result = await NPMPackage.create({ root }, { name: plugin })
```

Comment:
@ascorbic a bit late to the game but is there a reason this create call is awaited? It makes package installs serial — if you just return the promise, the NPMPackage provider will batch up all calls made at the same time into one npm/yarn install call.

Reply:
@KyleAMathews Afaict, his should still work, because we're doing the calls to `installPluginPackage` [in parallel](https://github.com/gatsbyjs/gatsby/pull/27703/files/cc862b35d34205d9452830b4d450638fed3a3929#diff-03eff4833bbe8d566c0a16b03c690ff13c7ccef78a0f76fdb31fcf9c9f46670fR76) in an async function. It will "await", but the async function will have already returned the promise so they'll still run in parallel

Reply:
ahhh yeah 😅 async is hard. Carry on!

Reply:
When it messes with my head, I find it helpful to think of an async function as _starting_ with `return new Promise`...